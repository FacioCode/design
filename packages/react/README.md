# Facio Design System components for React

[![lerna](https://img.shields.io/badge/maintained%20with-lerna-cc00ff.svg)](https://lerna.js.org/)
[![npm package](https://img.shields.io/npm/v/@faciocode/react/latest.svg)](https://www.npmjs.com/package/@faciocode/react)

## 

### Getting started

Install and get started with Facio Design System components for React for web applications.

#### 1. How to install

If your project uses `react` and `react-dom` version `16.14.0` or higher, run:

```shell script
npm i --save @faciocode/react
```

The [`@faciocode/react` npm package](https://www.npmjs.com/package/@faciocode/react) installs all Facio Design System UI components for React that you need.

This command will use the latest stable release. Please, always use `latest` distribution channel.

##### Not a React-based project?

Install our style tokens separately:

```shell script
npm i --save @faciocode/styles
```

#### 2. How to load the fonts

This package does not provide "Open Sans" font, only its font family names.

Load the Open Sans font with `400`, `600`, `700` and `800` font weights:

```html
<link href="https://fonts.gstatic.com"
      rel="preconnect" />
<link href="https://fonts.googleapis.com/css2?family=Open+Sans:wght@400;600;700;800&display=swap"
      rel="stylesheet">
```

#### 3. How to use

A quick example to get you started:

(Note that we enclose all your application in a `ThemeProvider` component)

```jsx
import {
  CssBaseline,
  ThemeProvider
} from "@faciocode/react";

export const App = () => {
  <ThemeProvider>
    <CssBaseline />
    <ContainedButton color={"brand"}>
      Hello Button
    </ContainedButton>
  </ThemeProvider>
} 
```

Check our [Design System Storybook](http://storybook.facio.dev/) for all React components.

This package only works fine with the use of `<ThemeProvider />`, **essential** for applying styles correctly.

Please do not try to override Facio Design System theme palette or another style tokens. Use the Design System themes and components as-is.

### Additional information

- Please note the package peer dependencies `react >= 16.14.0` and `react-dom >= 16.14.0`;
- We recommend at least LTS Node version (`v14.15.4+`);
- If your project uses TypeScript, we need TypeScript `4.1.2` at least;

#### Bundled in this package

Installing `@faciocode/react` will also install the following packages:

- `@faciocode/mui-styles`
- `@faciocode/styles`
- `@material-ui/core`
- Other dependencies for styling.

You don't need to install them separately to use the Design System.

### Questions?

For how-to questions and other issues, please use the [issues section on our GitHub repo](https://github.com/faciocode/design/issues).

### Changelog

Just upgraded? Please read the [Changelog](https://github.com/FacioCode/design/blob/main/packages/react/CHANGELOG.md).
