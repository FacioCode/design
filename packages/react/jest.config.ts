import { baseConfig } from "../../jest.config";
import { compilerOptions } from "./tsconfig.json";
import { pathsToModuleNameMapper } from "ts-jest/utils";
import pkg from "./package.json";

const config = {
  ...baseConfig,
  displayName: pkg.displayName,
  moduleNameMapper: pathsToModuleNameMapper(compilerOptions.paths, {
    prefix: "<rootDir>/",
  }),
};

export default config;
