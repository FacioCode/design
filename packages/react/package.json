{
	"name": "@faciocode/react",
	"displayName": "FacioReact",
	"version": "5.1.2",
	"private": false,
	"description": "React components for Facio",
	"keywords": [
		"react",
		"components",
		"facio",
		"design",
		"ds",
		"components"
	],
	"author": {
		"email": "technology@facio.com",
		"name": "Facio",
		"url": "https://github.com/FacioCode"
	},
	"homepage": "https://github.com/FacioCode/design/tree/main/packages/react#readme",
	"license": "ISC",
	"directories": {
		"lib": "dist",
		"source": "src"
	},
	"files": [
		"dist",
		"README.md",
		"dist/index.d.ts"
	],
	"esmodule": "dist/es/index.js",
	"jsdelivr": "dist/umd/index.js",
	"jsnext:main": "dist/es/index.js",
	"main": "dist/index.js",
	"module": "dist/es/index.js",
	"source": "src/index.ts",
	"types": "dist/index.d.ts",
	"typings": "dist/index.d.ts",
	"umd:main": "dist/umd/index.js",
	"unpkg": "dist/umd/index.js",
	"repository": {
		"directory": "packages/react",
		"type": "git",
		"url": "https://github.com/FacioCode/design"
	},
	"bugs": {
		"url": "https://github.com/FacioCode/design/issues"
	},
	"peerDependencies": {
		"@types/react": "^17.0.0",
		"react": "^16.14.0 || ^17.0.1",
		"react-dom": "^16.14.0 || ^17.0.1"
	},
	"dependencies": {
		"@faciocode/mui-themes": "^5.1.2",
		"@faciocode/styles": "^5.1.2",
		"@material-ui/core": "deprecations",
		"@material-ui/lab": "4.0.0-alpha.57",
		"clsx": "1.1.1",
		"lodash": "4.17.20"
	},
	"scripts": {
		"build": "npm run clean && npm run compile",
		"build-storybook": "build-storybook -c ./.storybook --quiet -o ./dist-storybook",
		"chromatic": "chromatic --output-dir dist-storybook",
		"chromatic-ci": "npm run chromatic --exit-once-uploaded",
		"clean": "rm -rf dist/",
		"compile": "npm run rollup && npm run emitDeclaration",
		"emitDeclaration": "tsc --declaration --declarationDir dist --emitDeclarationOnly",
		"rollup": "rollup -c",
		"start-storybook": "start-storybook --no-manager-cache -c ./.storybook -p 9009",
		"test": "echo \"Todo: write tests\" && exit 0"
	},
	"publishConfig": {
		"access": "public"
	}
}
