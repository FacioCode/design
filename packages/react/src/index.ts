export {
  Alert,
  AlertProps,
  AlertTitle,
  AlertTitleProps,
} from "./Alert";
export { AppBar, AppBarProps } from "./AppBar";
export { default as BottomSheet, BottomSheetProps } from "./BottomSheet";
export { ButtonGroup, ButtonGroupProps } from "./Button";
export {
  ContainedButton,
  ContainedButtonProps,
  CloseButton,
  CloseButtonProps,
  LinkButton,
  LinkButtonProps,
  OutlinedButton,
  OutlinedButtonProps,
} from "./Button";
export { Card, CardContent, CardProps } from "./Card";
export { Checkbox, CheckboxProps } from "./Checkbox";
export { Chip, ChipProps } from "./Chip";
export { default as Container, ContainerProps } from "./Container";
export { CssBaseline } from "./CssBaseline";
export { Currency, CurrencyProps } from "./Currency";
export {
  Dialog,
  DialogActions,
  DialogActionsProps,
  DialogContent,
  DialogContentProps,
  DialogContentText,
  DialogContentTextProps,
  DialogProps,
  DialogTitle,
  DialogTitleProps,
} from "./Dialog";
export { Divider, DividerProps } from "./Divider";
export { default as FormLabel, FormLabelProps } from "./FormLabel";
export { default as FormHelperText, FormHelperTextProps } from "./FormHelperText";
export { IconButton, IconButtonProps } from "./IconButton";
export { Input, InputProps } from "./Input";
export { default as InputBase, InputBaseProps } from "./InputBase";
export { default as InputLabel, InputLabelProps } from "./InputLabel";
export { default as LinearProgress, LinearProgressProps } from "./LinearProgress";
export { Link, LinkProps } from "./Link";
export { default as Paper, PaperProps } from "./Paper";
export { default as Popover, PopoverProps } from "./Popover";
export { default as Skeleton, SkeletonProps } from "./Skeleton";
export { Snackbar, SnackbarProps } from "./Snackbar";
export { default as Switch, SwitchProps } from "./Switch";
export {
  Table,
  TableBody,
  TableBodyProps,
  TableCell,
  TableCellProps,
  TableContainer,
  TableContainerProps,
  TableFooter,
  TableFooterProps,
  TableHead,
  TableHeadProps,
  TableProps,
  TableRow,
  TableRowProps,
} from "./Table";
export {
  default as Tabs,
  Tab,
  TabProps,
  TabPanel,
  TabPanelProps,
  TabsProps,
} from "./Tabs";
export { default as Toolbar, ToolbarProps } from "./Toolbar";
export { TextField, TextFieldProps } from "./TextField";
export { ThemeProvider, ThemeProviderProps } from "./ThemeProvider";
export { default as Time, TimeProps } from "./Time";
export { default as Tooltip, TooltipProps } from "./Tooltip";
export { Typography, TypographyProps } from "./Typography";

export { default as useMediaQuery } from "./hooks/useMediaQuery";
export { default as useTheme } from "./hooks/useTheme";

export { default as ServerStyleSheets } from "./ServerStyleSheets";

export { default as CloseIcon } from "./icons/CloseIcon";
export { default as WhatsAppIcon } from "./icons/WhatsAppIcon";
