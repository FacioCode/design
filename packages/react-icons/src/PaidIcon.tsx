/* eslint-disable indent, max-len */
import * as React from "react";
import { SvgIcon, SvgIconProps } from "@material-ui/core";

/**
 * Useful for approve icon button or paid status
 *
 * @example
 * ```jsx
 * <PaidIcon titleAccess={"Useful for approve icon button or paid status"} />
 * ```
 */
export const PaidIcon = React.forwardRef<SVGSVGElement, SvgIconProps>(
  (props, ref) => (
    <SvgIcon {...props} ref={ref}
      xmlns="http://www.w3.org/2000/svg" width="24" height="24">
      <path fillRule="evenodd" d="M5.349 12.027a.5.5 0 0 0-.698.716l4.737 4.615a.5.5 0 0 0 .698 0l10.263-10a.5.5 0 0 0-.698-.716l-9.914 9.66-4.388-4.275z"/>
    </SvgIcon>
  ),
);

// eslint-disable-next-line no-process-env
if (process.env.NODE_ENV !== "production") {
  PaidIcon.displayName = "PaidIcon";
}
