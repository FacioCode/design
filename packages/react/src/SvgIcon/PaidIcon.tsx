import * as React from "react";
import { SvgIconProps, SvgIconRender } from "@svg-icons/SvgIcon.types";
import SvgIcon from "@svg-icons/SvgIcon";

const render : SvgIconRender = (props, ref) => (
  <SvgIcon {...props} ref={ref}>
    {/* eslint-disable-next-line max-len */}
    <path fillRule="evenodd" d="M5.349 12.027a.5.5 0 0 0-.698.716l4.737 4.615a.5.5 0 0 0 .698 0l10.263-10a.5.5 0 0 0-.698-.716l-9.914 9.66-4.388-4.275z"/>
  </SvgIcon>
);

export const PaidIcon = React.forwardRef<unknown, SvgIconProps>(render);
PaidIcon.displayName = "PaidIcon";

export default PaidIcon;
