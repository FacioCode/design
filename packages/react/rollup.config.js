import { unpkg as browser, directories, displayName, module, source } from "./package.json";
import config from "../../config/rollup/rollup.config";

export default config({
  browser,
  directories,
  displayName,
  globals: {
    "@faciocode/styles": "FacioStyles",
    "@material-ui/core": "MuiCore",
    "@material-ui/core/styles": "MuiCoreStyles",
    "@material-ui/lab": "MuiLab",
    lodash: "_",
    "prop-types": "PropTypes",
    react: "React",
  },
  module,
  source,
});
