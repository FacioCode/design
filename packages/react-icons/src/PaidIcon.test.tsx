/* eslint-disable sort-imports */
import * as React from "react";
import { PaidIcon } from "./PaidIcon";
import { create } from "react-test-renderer";

describe("<PaidIcon component", () => {
  it("should match snapshot", () => {
    const testRenderer = create(<PaidIcon />);

    expect(testRenderer).toMatchSnapshot();
  });
});
