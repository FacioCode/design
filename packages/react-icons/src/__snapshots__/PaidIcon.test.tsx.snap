// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaidIcon component should match snapshot 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root"
  focusable="false"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M5.349 12.027a.5.5 0 0 0-.698.716l4.737 4.615a.5.5 0 0 0 .698 0l10.263-10a.5.5 0 0 0-.698-.716l-9.914 9.66-4.388-4.275z"
    fillRule="evenodd"
  />
</svg>
`;
